### PR TITLE
Feature/update project status

### PR DESCRIPTION
**Pull request by frankyma**
Update Project Step dialog.  It's just a long scroll down for now.  The new schema is from SeedCompany/cord-api-v3#1038, hence the failing CI but the code is ready for review

### CODE REVIEW

File: src/api/projectSteps.ts [hunk 1]
```diff
@@ -0,0 +1,62 @@
+enum ProjectStatus {
+  InDevelopment = 'InDevelopment',
+  Pending = 'Pending',
+  Active = 'Active',
+  Stopped = 'Stopped',
+  Finished = 'Finished',
+}
+
+enum ProjectStep {
```
Comment (CarsonF) [inline src/api/projectSteps.ts:9]: Both of these enums can be replaced with the string literals from generated schema.

File: src/scenes/Projects/Overview/ProjectOverview.tsx [hunk 2]
```diff
@@ -293,6 +311,66 @@ export const ProjectOverview: FC = () => {
           )}
         </div>
       )}
+      <DialogForm<UpdateProjectInput>
```
Comment (CarsonF) [inline src/scenes/Projects/Overview/ProjectOverview.tsx:314]: Separate component/file please

File: src/api/projectSteps.ts (hunk 1 above)
Comment (frankyma) [reply src/api/projectSteps.ts]: Changed

File: src/scenes/Projects/Overview/ProjectOverview.tsx (hunk 2 above)
Comment (frankyma) [reply src/scenes/Projects/Overview/ProjectOverview.tsx]: Moved

File: src/scenes/Projects/Update/UpdateProjectDialog.tsx [hunk 3]
```diff
@@ -0,0 +1,89 @@
+import { Typography } from '@material-ui/core';
```
Comment (CarsonF) [inline src/scenes/Projects/Update/UpdateProjectDialog.tsx:1]: Can you rename the file so it matches the component name?

File: src/scenes/Projects/Update/UpdateProjectDialog.tsx [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+        },
+      }}
+      onSubmit={async (input) => {
+        await updateProject({ variables: { input } });
+      }}
+    >
+      <SubmitError />
+      <RadioField name="project.step">
+        <Typography variant="h4">In Development</Typography>
+        {projectStepToStatusMap.InDevelopment.map((step) => (
```
Comment (CarsonF) [inline src/scenes/Projects/Update/UpdateProjectDialog.tsx:47]: You could loop through the groups as well

File: src/scenes/Projects/Overview/ProjectOverview.tsx [hunk 5]
```diff
@@ -293,6 +301,10 @@ export const ProjectOverview: FC = () => {
           )}
         </div>
       )}
+      <UpdateProjectStepDialog
```
Comment (CarsonF) [inline src/scenes/Projects/Overview/ProjectOverview.tsx:304]: I think it makes more sense to put this next to the DataButton. Dialogs use a Portal so it doesn't matter where they go in JSX.

File: src/scenes/Projects/Update/UpdateProjectDialog.tsx (hunk 3 above)
Comment (frankyma) [reply src/scenes/Projects/Update/UpdateProjectDialog.tsx:1]: Done

File: src/scenes/Projects/Update/UpdateProjectDialog.tsx (hunk 4 above)
Comment (frankyma) [reply src/scenes/Projects/Update/UpdateProjectDialog.tsx]: Fixed, much cleaner 👌

File: src/scenes/Projects/Overview/ProjectOverview.tsx (hunk 5 above)
Comment (frankyma) [reply src/scenes/Projects/Overview/ProjectOverview.tsx]: I was thinking of using this dialog for the update dates and location too

Comment (CarsonF) [reply src/scenes/Projects/Overview/ProjectOverview.tsx]: It's called UpdateStep though?

Comment (frankyma) [reply src/scenes/Projects/Overview/ProjectOverview.tsx]: I renamed it `UpdateProjectDialog`

**Comment by CarsonF** [comment]
@frankyma The dialog was really long due to all the step options. I changed it to an autocomplete. What do you think?

**Comment by frankyma** [comment]
> @frankyma The dialog was really long due to all the step options. I changed it to an autocomplete. What do you think?

Seems like a good idea.  I was thinking we'd divide into columns but autocomplete is easier

**Pull request merged**